ios-swiftui/realm-template-apps-swiftui/TaskViews.swift

```diff
@@ -0,0 +1,141 @@
+import SwiftUI
+import RealmSwift
+
+/// Show a detail view of a task. User can edit or mark the task complete.
+struct TaskDetail: View {
+    // This property wrapper observes the Task object and
+    // invalidates the view when the Task object changes.
+    @ObservedRealmObject var task: Task
+    
+    var body: some View {
+        Form {
+            Section(header: Text("Edit Task Summary")) {
+                // Accessing the observed task object lets us update the live object
+                // No need to explicitly update the object in a write transaction
+                TextField("Summary", text: $task.summary)
+            }
+            Section {
+                Toggle(isOn: $task.isComplete) {
+                    Text("Complete")
+                }
+            }
+        }
+        .navigationBarTitle("Update Task", displayMode: .inline)
+    }
+}
+
+struct TaskRow: View {
+    @ObservedRealmObject var task: Task
+    
+    var body: some View {
+        NavigationLink(destination: TaskDetail(task: task)) {
+            Text(task.summary)
+            Spacer()
+            if task.isComplete {
+                Image(systemName: "checkmark")
+                    .foregroundColor(.blue)
+                    .padding(.trailing, 10)
+            }
+        }
+    }
+}
+
+/// Use views to see a list of all Tasks, add or delete tasks, or logout.
+struct TasksView: View {
+    // ObservedResults is a mutable collection; here it's
+    // all of the Task objects in the realm.
+    // You can append or delete tasks directly from the collection.
+    @ObservedResults(Task.self) var tasks
+    @State var addTask = false
+    @State var isLoggingOut = false
```

Comment:
If you have already created views to handle logging out, there should be no reason for this view to also maintain state about logging in or out.